lib/retry.go

```diff
@@ -0,0 +1,20 @@
+package lib
+
+import (
+	"context"
+	"errors"
+	"time"
+)
+
+func WithRetries(ctx context.Context, retryFn func() error, options struct {
+	retriesCount    int
+	retriesInterval int
+}) error {
+	for i := 0; i < options.retriesCount; i++ {
+		if err := retryFn(); err == nil {
+			return nil
+		}
+		time.Sleep(time.Duration(options.retriesInterval) * time.Second)
```

Comment:
Caso o contexto seja cancelado ou o tempo limite tenha sido ultrapassado (i.e. `context.Canceled`, `context.DeadlineExceeded`), o ideal é que essa função pare de ser executada. Faz sentido passar o contexto aqui, e também no contrato da função retentada (pra que seja possível esse controle dentro dela também):
```suggestion
		if err := retryFn(ctx); err == nil {
			return nil
		}
		if err := ctx.Err(); err != nil {
		        return err
		}
		time.Sleep(time.Duration(options.retriesInterval) * time.Second)
```